yb-voyager/src/compareperf/comparison_test.go

```diff
@@ -0,0 +1,269 @@
+			RowsProcessed:   QUERY_CALLS_FOR_TEST,
+			TotalExecTime:   avgTime * float64(QUERY_CALLS_FOR_TEST),
+			AverageExecTime: avgTime,
+		})
+	}
+
+	// Generate target-only queries
+	for i := 0; i < targetOnly; i++ {
+		queryText := fmt.Sprintf("TargetQuery-%d", i+1)
+		queryID := int64(2000 + i + 1) // Use different ID range
+		avgTime := 8.0                 // Different from matched queries
+
+		targetStats = append(targetStats, &types.QueryStats{
+			QueryID:         queryID,
+			QueryText:       queryText,
+			ExecutionCount:  QUERY_CALLS_FOR_TEST,
+			RowsProcessed:   QUERY_CALLS_FOR_TEST,
+			TotalExecTime:   avgTime * float64(QUERY_CALLS_FOR_TEST),
+			AverageExecTime: avgTime,
+		})
+	}
+
+	return &QueryPerformanceComparator{
+		SourceQueryStats: sourceStats,
+		TargetQueryStats: targetStats,
+		msr:              nil, // not required as of now for tests
+		targetDB:         nil, // not required as of now for tests
+	}
+}
+
+// TestGetTopBySlowdown tests the getTopBySlowdown method with predictable mock data
+func TestGetTopBySlowdown(t *testing.T) {
+	// Create comparator with 30 matched queries, target 50% slower
+	comparator := createMockComparator(30, 10, 10, 1.5) // 1.5 = 50% slower
+	allComparisons := comparator.matchQueries()
+	for _, comp := range allComparisons {
+		comp.calculateMetrics()
+	}
+
+	// All matched queries should have same slowdown ratio since same slowdown factor (source avg time is 10ms)
+	expectedSlowdownRatio := (15.0 + SLOWDOWN_RATIO_OFFSET) / (10.0 + SLOWDOWN_RATIO_OFFSET)
+
+	topBySlowdown := comparator.getTopBySlowdown(allComparisons, 10)
+	assert.Len(t, topBySlowdown, 3, "Should return only the 3 matched queries")
+	for i, comp := range topBySlowdown {
+		assert.Equal(t, MATCHED, comp.MatchStatus, "Query %d should be MATCHED", i)
+		assert.InDelta(t, expectedSlowdownRatio, comp.SlowdownRatio, 0.001)
+		assert.Contains(t, comp.Query, "Query-", "Should be a matched query")
+	}
+
+	for i := 1; i < len(topBySlowdown); i++ {
+		assert.True(t, topBySlowdown[i-1].SlowdownRatio >= topBySlowdown[i].SlowdownRatio,
+			"Results should be sorted by slowdown ratio in descending order")
+	}
+}
+
+// TestGetTopBySlowdown_WithNLimit tests the limit parameter behavior
+func TestGetTopBySlowdown_WithTopNLimit(t *testing.T) {
+	// Create comparator with 50 matched queries, target 2x slower
+	comparator := createMockComparator(50, 0, 0, 2.0)
+	expectedSlowdownRatio := (20.0 + SLOWDOWN_RATIO_OFFSET) / (10.0 + SLOWDOWN_RATIO_OFFSET)
+	allComparisons := comparator.matchQueries()
+	for _, comp := range allComparisons {
+		comp.calculateMetrics()
+	}
+
+	topN := 3
+	topBySlowdown := comparator.getTopBySlowdown(allComparisons, topN)
+
+	// assertions
+	assert.Len(t, topBySlowdown, topN, "Should return only 3 results due to limit")
+	for _, comp := range topBySlowdown {
+		assert.Equal(t, MATCHED, comp.MatchStatus, "All results should be MATCHED")
+		assert.InDelta(t, expectedSlowdownRatio, comp.SlowdownRatio, 0.001)
+	}
+}
```

Comment:
In `TestGetTopBySlowdown` we are anyways providing and checking with a limit of 10. Is this separate test functionally adding anything new?